lib/features/auth/pages/email_signup_page.dart

```diff
@@ -0,0 +1,154 @@
+// Here's the code block for the page with an input field, register button, and forgot password button using Flutter/Dart:
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:givt_app/features/auth/pages/login_page.dart';
+import 'package:givt_app/l10n/l10n.dart';
+
+class EmailSignupPage extends StatelessWidget {
+  const EmailSignupPage({super.key});
+
+  static CupertinoPageRoute<dynamic> route() {
+    return CupertinoPageRoute(
+      fullscreenDialog: true,
+      builder: (_) => const EmailSignupPage(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return const InputPage();
+  }
+}
+
+class InputPage extends StatefulWidget {
+  const InputPage({super.key});
+
+  @override
+  _InputPageState createState() => _InputPageState();
+}
+
+class _InputPageState extends State<InputPage> {
+  final _formKey = GlobalKey<FormState>();
+  final TextEditingController _emailController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Wouldn't it be better if you split some of these long code blocks? Will make it more readable later if you need to lookup a specific text, button etc. 
This also applies to some other files.